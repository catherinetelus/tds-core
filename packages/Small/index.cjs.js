const Small = require('./dist/index.cjs')

module.exports = Small
