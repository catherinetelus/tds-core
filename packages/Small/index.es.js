import Small from './dist/index.es'

export default Small
