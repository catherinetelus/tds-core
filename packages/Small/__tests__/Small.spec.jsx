import React from 'react'
import { shallow, render } from 'enzyme'

import Small from '../Small'

describe('Small', () => {
  const doShallow = props => shallow(<Small {...props}>Some content</Small>)
  const doRender = props => render(<Small {...props}>Some content</Small>)

  it('renders', () => {
    const small = doRender()

    expect(small).toMatchSnapshot()
  })

  it('renders an HTML small tag', () => {
    const small = doRender()

    expect(small.is('small')).toEqual(true)
  })

  it('passes additional attributes to the small element', () => {
    const small = doShallow({ id: 'the-id' })

    expect(small).toHaveProp('id', 'the-id')
  })

  it('does not allow custom CSS', () => {
    const small = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(small).not.toHaveProp('className', 'my-custom-class')
    expect(small).not.toHaveProp('style')
  })
})
