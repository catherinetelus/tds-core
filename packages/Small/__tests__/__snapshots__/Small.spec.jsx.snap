// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Small renders 1`] = `
.c0 {
  word-wrap: break-word;
  font-size: 0.875rem;
  -webkit-letter-spacing: -0.6px;
  -moz-letter-spacing: -0.6px;
  -ms-letter-spacing: -0.6px;
  letter-spacing: -0.6px;
  line-height: 1.42857;
  font-weight: 500;
}

<small
  class="c0"
>
  Some content
</small>
`;
